diff -ruN a/target/linux/mediatek/image/mt7622.mk b/target/linux/mediatek/image/mt7622.mk
--- a/target/linux/mediatek/image/mt7622.mk	2024-12-18 18:00:00.104754420 +0800
+++ b/target/linux/mediatek/image/mt7622.mk	2024-12-09 14:33:04.161904403 +0800
@@ -292,6 +292,27 @@
 endef
 TARGET_DEVICES += mediatek_mt7622-rfb1-ubi
 
+define Device/asiarf_ap7622-wh1
+  DEVICE_VENDOR := AsiaRF
+  DEVICE_MODEL := AP7622 Wi-Fi HaLow Expert Kit
+  DEVICE_DTS := mt7622-asiarf-ap7622-wh1
+  DEVICE_DTS_DIR := ../dts
+  DEVICE_IMG_PREFIX := $$(DEVICE_NAME)-$$(IMG_PREFIX_VERCODE)
+  DEVICE_PACKAGES := kmod-ata-ahci-mtk kmod-btmtkuart kmod-usb3
+  BOARD_NAME := asiarf,ap7622-wh1
+  UBINIZE_OPTS := -E 5
+  BLOCKSIZE := 128k
+  PAGESIZE := 2048
+  KERNEL_SIZE := 4194304
+  IMAGE_SIZE := 32768k
+  IMAGES += factory.bin
+  IMAGE/factory.bin := append-kernel | pad-to $$(KERNEL_SIZE) | append-ubi | \
+                check-size $$$$(IMAGE_SIZE)
+  IMAGE/sysupgrade.bin := sysupgrade-tar | append-metadata
+  SUPPORTED_DEVICES := asiarf,ap7622-wh1
+endef
+TARGET_DEVICES += asiarf_ap7622-wh1
+
 define Device/netgear_wax206
   $(Device/dsa-migration)
   DEVICE_VENDOR := NETGEAR
diff -ruN a/target/linux/mediatek/mt7622/base-files/etc/board.d/02_network b/target/linux/mediatek/mt7622/base-files/etc/board.d/02_network
--- a/target/linux/mediatek/mt7622/base-files/etc/board.d/02_network	2024-12-18 18:00:00.104754420 +0800
+++ b/target/linux/mediatek/mt7622/base-files/etc/board.d/02_network	2024-10-22 09:06:38.429967417 +0800
@@ -15,6 +15,7 @@
 	linksys,e8450-ubi|\
 	mediatek,mt7622-rfb1|\
 	mediatek,mt7622-rfb1-ubi|\
+	asiarf,ap7622-wh1|\
 	netgear,wax206|\
 	reyee,ax3200-e5|\
 	ruijie,rg-ew3200gx-pro)
diff -ruN a/target/linux/mediatek/mt7622/base-files/lib/upgrade/platform.sh b/target/linux/mediatek/mt7622/base-files/lib/upgrade/platform.sh
--- a/target/linux/mediatek/mt7622/base-files/lib/upgrade/platform.sh	2024-12-18 18:00:00.104754420 +0800
+++ b/target/linux/mediatek/mt7622/base-files/lib/upgrade/platform.sh	2024-12-09 13:54:08.234752905 +0800
@@ -37,6 +37,7 @@
 	dlink,eagle-pro-ai-m32-a1|\
 	elecom,wrc-x3200gst3|\
 	mediatek,mt7622-rfb1-ubi|\
+	asiarf,ap7622-wh1|\
 	netgear,wax206|\
 	totolink,a8000ru|\
 	xiaomi,redmi-router-ax6s)
@@ -76,6 +77,7 @@
   	dlink,eagle-pro-ai-m32-a1|\
 	elecom,wrc-x3200gst3|\
 	mediatek,mt7622-rfb1-ubi|\
+	asiarf,ap7622-wh1|\
 	netgear,wax206|\
 	totolink,a8000ru|\
 	xiaomi,redmi-router-ax6s)

