--- a/mac.c	2024-10-14 16:29:27.532215692 +0800
+++ b/mac.c	2024-10-14 16:31:42.902374287 +0800
@@ -479,6 +479,18 @@
 
 };
 
+static const struct ieee80211_tpt_blink morse_tpt_blink[] = {
+    { .throughput =   0 * 1024, .blink_time = 334 },
+    { .throughput =   1 * 1024, .blink_time = 260 },
+    { .throughput =   5 * 1024, .blink_time = 220 },
+    { .throughput =  10 * 1024, .blink_time = 190 },
+    { .throughput =  15 * 1024, .blink_time = 170 },
+    { .throughput =  30 * 1024, .blink_time = 150 },
+    { .throughput =  50 * 1024, .blink_time = 130 },
+    { .throughput =  80 * 1024, .blink_time = 110 },
+    { .throughput = 100 * 1024, .blink_time =  80 },
+    { .throughput = 150 * 1024, .blink_time =  50 },
+};
 
 /* Returns true if only STA mode is supported */
 static bool is_sta_mode_only(void)
@@ -6515,6 +6527,12 @@
 	mors = hw->priv;
 	mors->hw = hw;
 
+    //Create led tpt
+    ieee80211_create_tpt_led_trigger(hw,
+                    IEEE80211_TPT_LEDTRIG_FL_CONNECTED,
+                    morse_tpt_blink,
+                    ARRAY_SIZE(morse_tpt_blink));
+
 	return mors;
 }
 
