diff --git a/feeds/morse/hardware/morse-bundle/files/etc/uci-defaults/98-morse-wireless-defaults b/hardware/morse-bundle/files/etc/uci-defaults/98-morse-wireless-defaults
--- a/feeds/morse/hardware/morse-bundle/files/etc/uci-defaults/98-morse-wireless-defaults
+++ b/feeds/morse/hardware/morse-bundle/files/etc/uci-defaults/98-morse-wireless-defaults
@@ -14,10 +14,12 @@
 . /lib/functions/system.sh
 
 
-default_ssid=$(uci get system.@system[0].hostname 2>/dev/null)
+default_ssid_prefix=$(uci get system.@system[0].hostname 2>/dev/null | cut -d '-' -f 1)
+default_ssid_suffix=$(uci get system.@system[0].hostname 2>/dev/null | cut -d '-' -f 2)
 default_wifi_key="$(persistent_vars_storage.sh READ default_wifi_key)"
 if [ -z "$default_wifi_key" ]; then
-	default_wifi_key=$(tr -dc 'A-Za-z0-9' < /dev/urandom | head -c 8)
+	#default_wifi_key=$(tr -dc 'A-Za-z0-9' < /dev/urandom | head -c 8)
+	default_wifi_key=12345678
 fi
 uci set "system.@system[0].default_wifi_key=$default_wifi_key"
 mm_region="$(persistent_vars_storage.sh READ mm_region)"
@@ -80,6 +82,16 @@ _change_device_defaults() {
 			fi
 
 			uci set "wireless.$config.channel=$channel"
+
+			uci batch <<- END
+				set wireless.default_$config.ssid='$default_ssid_prefix-HaLow-$default_ssid_suffix'
+				set wireless.default_$config.key='$default_wifi_key'
+				set wireless.default_$config.encryption='sae'
+				set wireless.default_$config.max_inactivity='300'
+				set wireless.default_$config.dtim_period='1'
+				set wireless.default_$config.ieee80211w='2'
+			END
+
 			echo "uci-default-morse-wireless: setting $config to $mm_region on $channel" > /dev/kmsg
 		fi
 	elif [ "$type" = mac80211 ] &&
@@ -98,7 +110,7 @@ _change_device_defaults() {
 
 	uci batch <<- END
 		delete wireless.$config.disabled
-		set wireless.default_$config.ssid='$default_ssid'
+		set wireless.default_$config.ssid='$default_ssid_prefix-WiFi-$default_ssid_suffix'
 		set wireless.default_$config.key='$default_wifi_key'
 	END
 }
-- 
2.34.1

